### ISSUE TITLE
[deployer] Email notification enhancements

### ISSUE THREAD

**Issue by avasquez614** [craftercms/craftercms#5588]
#### Is your feature request related to a problem? Please describe.
Deployer email notifications are great but they need some work.

#### Describe the solution you'd like
- The Deployer should use the `MAIL_*` properties from the `crafter_setenv.sh` for sending emails
- Don't attach the full deployments CSV file when sending a notification. 
- `updateDetails` and `updateLog` properties don't seem necessary in serialization.
- The management token shouldn't be printed

**Comment by jdrossl** [comment]
@avasquez614 It seems like the first point is already working as expected https://github.com/craftercms/craftercms/blob/0e256ef0372c7be9d6e2fefc4652dd4fd94770a1/resources/deployer/config/base-target.yaml#L45

Let me know if you had a specific issue with that, it is working for me

**Comment by avasquez614** [comment]
Additional changes:

- Seems we have an issue with the CSV, the second file list should be `updated_files`, not `deleted_files`: https://github.com/craftercms/deployer/blob/develop/src/main/java/org/craftercms/deployer/impl/processors/FileOutputProcessor.java#L47
- No need to send a CSV on deployment email, seems redundant. The JSON attached is more than enough. The CSV doesn't seem like it's going to be used.
- Allow specifying in the `base-target.yaml` the following notification email properties (`properties` should be accepted as a dynamic map if possible):
```
    # The username of the email server
    username: ...
    # The password of the email server
    password: ...
    # Additional properties for the email server
    properties:
      mail.smtp.auth: ...
```

**Comment by jdrossl** [comment]
SMTP properties need to be expressed as a list of string:

```yaml
properties:
  - 'mail.smtp.auth=true'
```

**Comment by yacdaniel** [comment]
Verified fixed in `4.0.1-SNAPSHOT`
